File: src/MintADB.Android/app/src/main/java/com/minthd/mintadb/core/DebloatBlacklist.kt
package com.minthd.mintadb.core

object DebloatBlacklist {
    private val reasons = mapOf(
        "com.lbe.security.miui" to "Permission manager — bootloop risk",
        "com.android.updater" to "System updater — bootloop risk",
        "com.miui.securitycenter" to "Mi Security — bootloop risk",
        "com.miui.securityadd" to "Mi Security add-on",
        "com.xiaomi.finddevice" to "Find device — bootloop risk",
        "com.miui.home" to "System launcher — bootloop risk",
        "com.miui.guardprovider" to "MIUI security component",
        "com.xiaomi.market" to "GetApps / Xiaomi store",
        "com.xiaomi.account" to "Xiaomi account",
        "com.miui.packageinstaller" to "MIUI package installer",
    )

    fun isProtected(packageName: String): Boolean = packageName in reasons

    fun getReason(packageName: String): String? = reasons[packageName]
}

File: src/MintADB.Wpf/Helpers/DebloatBlacklist.cs
namespace MintADB.Wpf.Helpers;

/// <summary>
/// Trước đây chặn gỡ một số gói hệ thống (Security/Updater/Launcher…).
/// Hiện không chặn — người dùng tự chịu rủi ro bootloop nếu gỡ nhầm.
/// </summary>
public static class DebloatBlacklist
{
    public static bool IsProtected(string package) => false;

    public static string? GetReason(string package) => null;

    public static IReadOnlyList<string> FilterProtected(IEnumerable<string> packages) => [];
}
